### PR TITLE
Adjust imports to updates in `libpysal`

### LINKED ISSUE #271: Access packages directly, not through `libpysal.common`
xref https://github.com/pysal/libpysal/issues/605

This is causing failures in:
* [`giddy`](https://github.com/pysal/giddy/actions/runs/6620883811/job/17984038516) 
* [`spaghetti`](https://github.com/pysal/spaghetti/actions/runs/6620730952/job/17983660632)
* [`splot`](https://github.com/pysal/splot/actions/runs/6620794945/job/17983814825)
* [`spopt`](https://github.com/pysal/spopt/actions/runs/6621375801/job/17985247353?pr=406)
* [`spreg`](https://github.com/pysal/spreg/actions/runs/6620815159/job/17983865929)

### PR DESCRIPTION
**Pull request by jGaboardi**
* This PR resolves #271.
* xref https://github.com/pysal/libpysal/issues/605

### CODE REVIEW

File: esda/getisord.py [hunk 1]
```diff
@@ -6,7 +6,9 @@
 
 import warnings
 
-from libpysal.common import np, stats
+import numpy as np
+import scipy.stats as stats
```
Comment (martinfleis) [inline esda/getisord.py:10]:
```suggestion
from scipy import stats
```

**Review by martinfleis — COMMENTED** [review]
This is an interesting pattern...

File: esda/getisord.py (hunk 1 above)
Comment (jGaboardi) [reply esda/getisord.py:10]: Yeah, [legacy from `libpysal.common`](https://github.com/pysal/libpysal/pull/597/files#diff-bfc26b652705b9cd76aed293a3c7f0eae9e87bbb92c2ffc55eb6cc576b8ac6c2L10).

**Review by martinfleis — APPROVED** [review]

**Comment by martinfleis** [comment]
We will need to release this before the next lib will come. Just a note.

**Comment by jGaboardi** [comment]
Any blockers at the moment? No time like the present.

**Comment by martinfleis** [comment]
We could easily do a patch just with this but also can wait. I don't assume lib will have a next release before end of the year.

**Comment by jGaboardi** [comment]
So would it hurt to have an `esda=2.5.1` release? I can trigger that if so. But also, would it hurt to have a `libpysal=4.8.1`?

**Comment by martinfleis** [comment]
> So would it hurt to have an esda=2.5.1 release?

no, better be ready.

> But also, would it hurt to have a libpysal=4.8.1?

no need for that, we're not fixing anything a user would care about

**Comment by jGaboardi** [comment]
Done.

**Pull request merged**